tests/tests_pytorch/profilers/test_hpu_profiler.py

```diff
@@ -0,0 +1,224 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import json
+import os
+import shutil
+
+import pytest
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.accelerators import HPUAccelerator
+from pytorch_lightning.demos.boring_classes import BoringModel
+from pytorch_lightning.profilers import AdvancedProfiler, HPUProfiler, SimpleProfiler
+from tests_pytorch.helpers.runif import RunIf
+
+
+class TestHPUProfiler:
+    def setup_method(self):
+        shutil.rmtree("profiler_logs", ignore_errors=True)
+
+    def teardown_method(self):
+        shutil.rmtree("profiler_logs", ignore_errors=True)
+
+    @pytest.fixture
+    def get_device_count(self, pytestconfig):
+        hpus = int(pytestconfig.getoption("hpus"))
+        if not hpus:
+            assert HPUAccelerator.auto_device_count() >= 1
+            return 1
+        assert hpus <= HPUAccelerator.auto_device_count(), "More hpu devices asked than present"
+        assert hpus == 1 or hpus % 8 == 0
+        return hpus
+
+    @RunIf(hpu=True)
+    def test_hpu_simple_profiler_instances(self, tmpdir, get_device_count):
+        model = BoringModel()
+        trainer = Trainer(
+            profiler="simple", accelerator="hpu", devices=get_device_count, max_epochs=1, default_root_dir=tmpdir
+        )
+        assert isinstance(trainer.profiler, SimpleProfiler)
+        trainer.fit(model)
```

Comment:
this one doesn't need a fit call

Reply:
doesn't hurt to check if fit works with profiling